### PR TITLE
Add documentation with Jupyterlite demo

### PR DESCRIPTION
**Pull request by brichet**
This adds a first documentation including a jupyterlite demo.

Wait for https://github.com/jupyterlite/jupyterlite-sphinx/pull/108 to be merged.

Fixes https://github.com/jupyterlite/litegitpuller/issues/6

### CODE REVIEW

**Comment by brichet** [comment]
Looks ready, but to try it in the built documentation of the PR, the src path of the iframe must be change from `src=/lite/lab` to `src=lite/lab` (removing the first slash).
I assume it only occurs on the PR documentation.

To test the git puller feature, this link can be used https://litegitpuller--8.org.readthedocs.build/en/8/?repo=https%3A%2F%2Fgithub.com%2Fbrichet%2Ftesting-repo&urlpath=tree%2Ftesting-repo%2Fnotebooks%2Fsimple.ipynb&branch=main, but the first slash of the iframe src need to be removed in the html source.

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (jtpio) [inline docs/Makefile:1]: Normally we should be able to avoid using a `Makefile`, to make things simpler.

Comment (jtpio) [reply docs/Makefile:1]: For example we could follow the `voici` setup and use hatch scripts: https://github.com/voila-dashboards/voici/blob/1587e7bfa2a87b71be568a20e385dd9d8da6e2ae/pyproject.toml#L135-L140

Comment (brichet) [reply docs/Makefile]: Thanks, I updated the PR.
Can you explain when this additional hatch script for documentation is triggered ?
As far as I understand readthedoc build the documentation itself for PRs and branch.

**Comment by brichet** [comment]
@jtpio does the RTD preview action is usually working on voici https://github.com/voila-dashboards/voici/blob/1587e7bfa2a87b71be568a20e385dd9d8da6e2ae/.github/workflows/rtd-preview.yml ?
It does not work on the current opened PR, neither on some closed I tried.

**Comment by jtpio** [comment]
Yes, for example: https://github.com/voila-dashboards/voici/pull/89#issuecomment-1688518032

But the workflow would first have to be merged and available on `main`.

**Comment by brichet** [comment]
> the src path of the iframe must be change from `src=/lite/lab` to `src=lite/lab` (removing the first slash)

This is caused by https://github.com/jupyterlite/jupyterlite-sphinx/pull/108.
I'll open a PR to fix it.

**Comment by brichet** [comment]
> Yes, for example: [voila-dashboards/voici#89 (comment)](https://github.com/voila-dashboards/voici/pull/89#issuecomment-1688518032)

The badge is visible but it leads me to a 404.

**Comment by brichet** [comment]
> > the src path of the iframe must be change from `src=/lite/lab` to `src=lite/lab` (removing the first slash)
> 
> This is caused by https://github.com/jupyterlite/jupyterlite-sphinx/pull/108.
> I'll open a PR to fix it.

Fixed

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
We can double check the `Check Links` job passes correctly on CI once merged into `main`.

**Pull request merged**